examples/basic_authentication/lib/user_repository.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:equatable/equatable.dart';
+import 'package:meta/meta.dart';
+
+final List<User> _db = [];
+
+@visibleForTesting
+
+/// Used to mock the database of the repository.
+void mockUserData(List<User> users) {
+  _db
+    ..clear()
+    ..addAll(users);
+}
+
+@visibleForTesting
+
+/// Used in tests to get the data from the memory database.
+List<User> getDb() => _db;
```

Comment:
I think it makes more sense to just make `_db` visible for testing.

Reply:
Why? Doesn't that accomplish the same thing?

Reply:
The idea behind it is allow moving the test related methods to the test instead of letting them be in the source code. For example, I would expect a method named `mockUserData` to be inside a test file.